Assets/Scripts/Models/Furniture.Factory.cs

```diff
@@ -0,0 +1,185 @@
+                }
+            }
+            else
+            {
+                // processing is in progress
+                furnParameters[CUR_PROCESSING_TIME_PARAM_NAME].ChangeFloatValue(deltaTime);
+
+                if (furnParameters[CUR_PROCESSING_TIME_PARAM_NAME].ToFloat() >=
+                    furnParameters[MAX_PROCESSING_TIME_PARAM_NAME].ToFloat())
+                {
+                    List<TileObjectTypeAmount> outPlacement = new List<TileObjectTypeAmount>();
+                    // processing is done, try to spit the output
+                    // check if output can be placed in world
+
+                    foreach (var outObjType in cChain.Output)
+                    {
+                        int amount = outObjType.Amount;
+
+                        // check ouput slots for products:                        
+                        Tile tt = World.Current.GetTileAt(
+                            Tile.X + outObjType.SlotPosX, Tile.Y + outObjType.SlotPosY);
+
+                        if (tt.Inventory == null || tt.Inventory.objectType == outObjType.ObjectType
+                            && tt.Inventory.StackSize + amount <= tt.Inventory.maxStackSize)
+                        {
+                            // out product can be placed here
+                            outPlacement.Add(new TileObjectTypeAmount()
+                            {
+                                Tile = tt,
+                                IsEmpty = tt.Inventory == null,
+                                ObjectType = outObjType.ObjectType,
+                                Amount = outObjType.Amount
+                            });
+                        }
+                    }
+
+                    // if output placement was found for all products, place them
+                    if (outPlacement.Count == cChain.Output.Count)
+                    {
+                        foreach (var outPlace in outPlacement)
+                        {
+                            if (outPlace.IsEmpty)
+                                World.Current.inventoryManager.PlaceInventory(outPlace.Tile,
+                                    new Inventory(outPlace.ObjectType, outPlace.Amount));
+                            else
+                                outPlace.Tile.Inventory.StackSize += outPlace.Amount;
+                        }
+                        // processing done, can fetch input for another processing
+                        furnParameters[CUR_PROCESSED_INV_PARAM_NAME].SetValue(0);
+                    }
+                }
+            }
+
+            // create possible jobs for factory (hauling input)
+            // - if input slot is empty pick whatever material factory can process (that has highest priority?)
+            // - if input slot already contains some inventory, gather more from that type until stack is full
+
+
+
+            // TODO: can cause problems if there are more inputs and 1 is hanging
+            if (JobCount() > 0)
```

Comment:
This should have brackets `{}` and probably a warning or error log if it shouldn't happen.